### ISSUE TITLE
Passing flattened tensor into RNN results in an error

### ISSUE THREAD

**Issue by Migdalin** [tensorflow/tensorflow#25421]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10

- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No

- TensorFlow installed from (source or binary): pip install tensorflow-gpu

- TensorFlow version (use command below):  b'v1.12.0-rc2-3-ga6d8ffae09' 1.12.0

- Python version: 3.6.5
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source):  N/A
- CUDA/cuDNN version:  9.0
- GPU model and memory:  GeForce RTX 2070

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
Passing flattened tensor into RNN results in an error:  "Shape (608, ?) must have rank at least 3"
More:  I have been trying to find some way to squash down my data via CNNs.  I keep hitting strange dimension issues, where TF complains about None.

**Describe the expected behavior**
I would like to pass my flattened data into either a Dense or an RNN layer.

**Code to reproduce the issue**

from collections import namedtuple
import tensorflow as tf
import numpy as np

ConvArgs = namedtuple('ConvArgs',
                      'layerInput, numFilters, filterSize, stride, init, namePrefix')
ConvArgs.__new__.__defaults__ = (None,) * len(ConvArgs._fields)

# Set to false to see shape output of the final, flattened layer.
showError = True

def BuildConv2D(convArgs):
    channelAxis = 3
    filterShape = [convArgs.filterSize, 
                   convArgs.filterSize, 
                   convArgs.layerInput.get_shape()[channelAxis], 
                   convArgs.numFilters]

    filters = tf.get_variable(shape=filterShape, 
                              dtype=tf.float32,
                              initializer=convArgs.init,
                              name=convArgs.namePrefix + 'filters')
    
    conv = tf.nn.conv2d(input=convArgs.layerInput,
                        filter=filters, 
                        strides=[1,convArgs.stride,convArgs.stride,1], 
                        padding='VALID')

    activated = tf.nn.relu(conv)
    return activated

tf.reset_default_graph()

with tf.Session() as sess:
    data = tf.placeholder(dtype=tf.float32, 
                          shape=(None,150000,1), 
                          name='data')

    dataShape = tf.shape(data)

    numCols = 80
    numRows = int(150000 / numCols)
    data2D = tf.reshape(data, (tf.shape(data)[0], numRows, numCols, 1))

    kernelInit = tf.orthogonal_initializer()
    conv1 = BuildConv2D(
            ConvArgs(layerInput = data2D, 
                     numFilters = 8,
                     filterSize = 8,
                     stride = 4,
                     init = kernelInit,
                     namePrefix='c1'))

    conv2 = BuildConv2D(
            ConvArgs(layerInput = conv1, 
                     numFilters = 16,
                     filterSize = 8,
                     stride = 4,
                     init = kernelInit,
                     namePrefix='c2'))

    conv3 = BuildConv2D(
            ConvArgs(layerInput = conv2, 
                     numFilters = 16,
                     filterSize = 3,
                     stride = 3,
                     init = kernelInit,
                     namePrefix='c3'))

    c3_flattened = tf.layers.Flatten()(conv3)
 
    if(showError == True):
        lstmCell = tf.contrib.rnn.LSTMBlockCell(num_units=16)
    
        rawOutputs, final_state = tf.nn.dynamic_rnn(
                lstmCell, 
                c3_flattened, 
                dtype=tf.float32)

    sess.run(tf.global_variables_initializer())
    result = sess.run(c3_flattened, feed_dict={data:  np.random.randint(-5, 12, size=(2,150000,1))})
    print(result.shape)
    print(result[:, 0:20])
    
    if(showError == True):
        resultA, resultB = sess.run([rawOutputs, final_state], feed_dict={data:  np.random.randint(-5, 12, size=(2,150000,1))})
        print(resultA.shape)
        print(resultB.shape)

**Other info / logs**
When attempting to use a Dense layer, I ran into the issue described here:
https://github.com/tensorflow/tensorflow/issues/13348

**Comment by qlzh727** [comment]
Sorry for the late reply.

If I read your code correctly, the input to LSTM is the output from a flatten layer, which should be a 2D tensor. If that's the case, then it is expected to throw an error since RNN will expect a 3D tensor as input (batch, timestep, feature).

**Comment by qlzh727** [comment]
Closing this issue for now since the input the does not meet the expectation for RNN. Please reopen the issue if you feel this is not the case.
